server/controller/src/main/java/ai/starwhale/mlops/domain/model/ModelService.java

```diff
@@ -266,6 +292,23 @@ private ModelInfoVo toModelInfoVo(ModelEntity model,
         }
     }
 
+    private List<FileNode> parseManifestFiles(String manifest) throws JsonProcessingException {
+        List<FileNode> files = new ArrayList<>();
+        if (StringUtils.hasText(manifest)) {
+            var meta = yamlMapper.readValue(manifest, MetaInfo.class);
+            for (MetaInfo.Resource fileDesc : meta.getResources()) {
+                var fork = FileNode.parse(fileDesc.getPath(), (node) -> {
+                    node.setSize(FileUtil.readableFileSize(fileDesc.getSize()));
+                    node.setDesc(fileDesc.getDesc().name());
+                    node.setSignature(fileDesc.getSignature());
+                });
+                // merge to files
+                FileNode.merge(files, fork);
+            }
+        }
+        return files;
```

Comment:
could we combine `parse` and `merge` to one method like `makeTree` which tasks `meta.getResources()` as a param.
Such then:
1. return `FileNode.makeTree` here
2. reuse `FileNode.makeTree`  when runtime diff or dataset diff

Reply:
done